### PR TITLE
Chromium: Implement User Agent support

### PR DESCRIPTION
**Pull request by zakharvoit**
Use Chromium backend's SessionSettings class to support settings user agent mode and user agent override.

depends on https://github.com/Igalia/wolvic-chromium/pull/19

### CODE REVIEW

File: app/src/common/chromium/com/igalia/wolvic/browser/api/impl/SettingsImpl.java [hunk 1]
```diff
@@ -73,7 +76,7 @@ public boolean getFullAccessibilityTree() {
 
     @Override
     public void setUserAgentMode(int value) {
-        // TODO: implement
+        mSessionSettings.setUserAgentMode(value);
```
Comment (jdapena) [inline app/src/common/chromium/com/igalia/wolvic/browser/api/impl/SettingsImpl.java:79]: As suggested in code review, we should not depend in chromium backend side on the specific values of the user agent mode defined in wolvic side.

Here we should convert the wolvic values to the chromium backend values.

File: app/src/common/chromium/com/igalia/wolvic/browser/api/impl/SettingsImpl.java [hunk 2]
```diff
@@ -88,8 +91,7 @@ public void setViewportMode(int value) {
 
     @Override
     public int getUserAgentMode() {
-        // TODO: implement
-        return WSessionSettings.USER_AGENT_MODE_MOBILE;
+        return mSessionSettings.getUserAgentMode();
```
Comment (jdapena) [inline app/src/common/chromium/com/igalia/wolvic/browser/api/impl/SettingsImpl.java:94]: Ditto, here we would convert from chromium backend values to wolvic values.

**Review by jdapena — CHANGES_REQUESTED** [review]

File: app/src/common/chromium/com/igalia/wolvic/browser/api/impl/SettingsImpl.java (hunk 2 above)
Comment (zakharvoit) [reply app/src/common/chromium/com/igalia/wolvic/browser/api/impl/SettingsImpl.java]: Fixed.

**Review by tiagovignatti — APPROVED** [review]
LGTM. Thank you!

**Review by jdapena — APPROVED** [review]

**Review by svillar — APPROVED** [review]
Nice

**Pull request merged**